React/wbooks/src/app/screens/BookDetail/index.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+
+import booksJson from '../../../constants/data';
+import reactSvg from '../../assets/default_book.svg';
+import ErrorIcon from '../../assets/sad_icon.png';
+
+import BookSummary from './components/BookSummary/index.js';
+import Suggestion from './components/Suggestion/index.js';
+import NewComment from './components/NewComment/index.js';
+import Comment from './components/Comment/index.js';
+import styles from './styles.scss';
+
+class BookDetail extends Component {
+  state = { book: '', error: '' };
+  backTitle = '<Volver';
+
+  componentWillMount() {
+    const book = booksJson.filter(book => { return book.id == this.props.match.params.bookId; });
+    if(book.length === 1){
+      this.setState({ book: book[0] });
+    }else if(book.length > 1){
+      this.setState({ error: 'Sorry, it was found more than one book with the given id.' });
+    }else {
+      this.setState({ error: 'Sorry, the book was not found.' });
+    }
+  }
+
+  render() {
+    if(this.state.error){
+      return (
+        <div className={styles.errorContainer}>
+          <img src={ErrorIcon} className={styles.errorImage} alt="error" />
+          <div className={styles.errorMessage}>{this.state.error}</div>
+        </div>
+      );
+    };
```

Comment:
This behavior applied here could be applied to the whole application. now I would leave it in this way, but think it as an extra that this behavior is used with a HOC, applied to the component where you use react-router, to handle the error of routes of the whole application